examples/sunlinsol/ginkgo/test_sunlinsol_ginkgo.cpp

```diff
@@ -171,12 +214,37 @@ void fill_matrix(gko::matrix::Dense<sunrealtype>* matrix)
   sunindextype mat_cols = matrix->get_size()[1];
   sunrealtype* mat_data = matrix->get_values();
 
-#if defined(USE_CUDA)
+#if defined(USE_CUDA) || defined(USE_HIP)
   unsigned threads_per_block = 256;
   unsigned num_blocks        = (mat_rows + threads_per_block - 1) / threads_per_block;
 
   fill_kernel<<<num_blocks, threads_per_block>>>(mat_rows, mat_cols, mat_data);
-  HIP_OR_CUDA(hipDeviceSynchronize();, cudaDeviceSynchronize(););
+  HIP_OR_CUDA(hipDeviceSynchronize(), cudaDeviceSynchronize(), matrix->get_executor()->synchronize());
```

Comment:
Since this line would only be compiled for HIP or CUDA, I think you can revert it back.